test/transformer/transformer_maps_test.dart

```diff
@@ -19,11 +18,22 @@ part "transformer_test_core.dart";
 /// We pretend that our main rule, [ThingWithDateSerializatinoRule] is generated
 /// so we can't or don't want to just modify it, so instead we subclass and
 /// modify the data before calling it.
-class SpecialThingWithDateRule extends ThingWithDateSerializationRule {
+class SpecialThingWithDateRule extends CustomRule {
+
+  CustomRule generated;
```

Comment:
If we also have the ability to run this at development-time, as discussed separately, that seems like it would address the editor error. You could even mix and match them, that is you could run the generation once, so the editor has not just the files but also the rules, but they aren't updating automatically. Then you could have the transformer update them.